src/script/sign.cpp

```diff
@@ -230,16 +230,16 @@ static bool SignTaproot(const SigningProvider& provider, const BaseSignatureCrea
     // Try key path spending.
     {
         KeyOriginInfo info;
-        if (provider.GetKeyOriginByXOnly(spenddata.internal_key, info)) {
-            auto it = sigdata.taproot_misc_pubkeys.find(spenddata.internal_key);
+        if (provider.GetKeyOriginByXOnly(sigdata.tr_spenddata.internal_key, info)) {
```

Comment:
37f6d29df5405d0cbebbb4368a6dcbc95c8f00e5 : maybe just squash this entire commit into f5e58b6b3dfa1622d7c03f82318ef06eb6c6818c or f7c7be9794d868a91c25a743f00b3fd351589d5f and add a comment along the lines of the commit message?

Reply:
IMO it's a separate enough change that it should be its own commit.